.github/workflows/test_on_pr.yml

```diff
@@ -64,46 +61,39 @@ jobs:
         ref: ${{ github.event.pull_request.head.sha }}
         fetch-depth: 0
 
-    - name: Setup Alpine Linux ${{matrix.size}}
+    - name: Setup Alpine Linux
       uses: jirutka/setup-alpine@v1
       if: ${{ matrix.size == '32b' }}
       with:
         arch: x86
         packages: >
-          go
           make 
           git 
           gcc 
           musl-dev
 
-    - name: getconf x86
-      if: ${{ matrix.size == '32b' }}
-      run: |
-        getconf LONG_BIT
-        go version
-      shell: alpine.sh {0}
-
-    - name: getconf x64
-      if: ${{ matrix.size == '64b' }}
-      run: |
-        getconf LONG_BIT
-        go version
-
     - name: Test without coverage (Windows, MacOS)
-      if: ${{matrix.os == 'macos-latest' || matrix.os == 'windows-latest'}}
+      if: ${{ matrix.os == 'macos-latest' || matrix.os == 'windows-latest' }}
       run: make test
 
     - name: Test without coverage (Ubuntu x86)
-      if: ${{matrix.os == 'ubuntu-latest' && matrix.size == '32b'}}
-      run: make test
+      if: ${{ matrix.size == '32b' }}
+      run: |
+        GOVERSION=${{ matrix.golang == '1.21' && '1.21.10' || '1.22.3' }}
```

Comment:
Please put 1.21.10 and 1.22.3 in the matrix, so you don't have to test here. This will most probably be overlooked the day somebody will adjust the versions in the matrix above, and then things will start to fail.